src/gui/PrivateBrowser.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ *  Copyright (C) 2023 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrivateBrowser.h"
+#include "MainWindow.h"
+#include "gui/osutils/OSUtils.h"
+#include <QProcess>
+
+void PrivateBrowser::openUrl(const QUrl& url)
```

Comment:
Do we need a whole class for this or can we get away with making this a GuiTools function?

Reply:
Maybe we could cleanup this and just use OSUtils for it.

Reply:
Yah just get the browser path from osutils and move the private mode argument code to database widget. Opening urls in general corks be moved to guitools eventually. We stuff way too much into database widget